library/dnet.c

```diff
@@ -433,12 +433,13 @@ static int dnet_iterator_server_send_complete(struct dnet_addr *addr, struct dne
 		err = cmd->status;
 
 		/*
-		 * Write CAS error is not a real 'error' in that regard, that we do not remove
-		 * local record, but also do not stop iterator.
-		 *
 		 * Setting @write_error forces iterator to stop.
+		 *
+		 * Stop iteration only if connection was failed on write
+		 * or no space left on remote backend.
 		 */
-		if (err && !send->write_error && (err != -EBADFD))
+		const int critical_error = err && (err == -ENXIO || err == -ENOSPC);
```

Comment:
Ok, we could skip it too.

I do know that for example write returns EINVAL in many cases and this should stop iterator.
Number of errors to skip is much smaller than that of to proceed, let's write them all in this place and add comments about cases when these errors happen, this will help with understanding what to expect from write.